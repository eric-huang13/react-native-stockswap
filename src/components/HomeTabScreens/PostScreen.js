import React, {useState} from 'react';
import {
  Text,
  View,
  StyleSheet,
  Image,
  SafeAreaView,
  TouchableOpacity,
  ScrollView,
  TextInput,
  Modal,
  KeyboardAvoidingView,
} from 'react-native';
import UserCommentList from './UserCommentList';
import ReportModal from '../HomeTabComponents/ReportModal';
import ShareToModal from '../HomeTabComponents/ShareToModal';
import LikeInactiveIcon from '../../icons/LikeInactiveIcon';
import CommentIcon from '../../icons/CommentIcon';
import {moderateScale} from '../../util/responsiveFont';
import MoreBox from '../HomeTabComponents/MoreBox';
import {
  EDIT_POST,
  REMOVE_POST,
  REPOST,
  COPY_LINK,
  SHARE_POST,
  TURN_ON_NOTIFICATIONS,
  REPORT,
  myPostsOptions,
  otherPostsOptions,
} from './constants';

const PostScreen = ({navigation, route}) => {
  const {post, userAccount} = route.params;
  const accountId = userAccount?.id;
  const [reportModalState, setReportModalState] = useState(false);
  const [shareModalState, setShareModalState] = useState(false);
  const [optionsModalState, setOptionsModalState] = useState(false);
  const optionsState =
    userAccount.id === post.userId ? myPostsOptions : otherPostsOptions;

  const reportModal = () => {
    setReportModalState(true);
  };

  const shareModal = () => {
    setShareModalState(true);
  };

  const navigationByCondition = () => {
    if (post.userId === accountId) {
      navigation.navigate({
        name: 'MyProfile',
        params: {id: post.id},
      });
    } else {
      navigation.navigate({
        name: 'Profile',
        params: {id: post.userId},
      });
    }
  };
  const dropDownSelect = (post, userAccount) => {
    // this.setState({shouldShow: false});
    navigation.navigate({
      name: 'EditPost',
      params: {post, userAccount},
    });
  };
  const onSelectOption = (action) => {
    switch (action) {
      case EDIT_POST:
        break;
      case REMOVE_POST:
        break;
      case REPOST:
        break;
      case COPY_LINK:
        break;
      case SHARE_POST:
        shareModal();
        break;
      case TURN_ON_NOTIFICATIONS:
        break;
      case REPORT:
        reportModal();
        break;
      default:
        console.log('Not a valid action');
    }
  };

  return (
    <SafeAreaView style={style.container}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        keyboardVerticalOffset={100}
        style={{flex: 1}}>
        <Modal
          transparent={true}
          visible={reportModalState}
          animationType="slide">
          <ReportModal
            reportModal={reportModal}
            onClose={() => setReportModalState(false)}
          />
        </Modal>
        <Modal
          transparent={true}
          visible={shareModalState}
          animationType="slide">
          <ShareToModal
            shareModal={setShareModalState}
            onClose={() => setShareModalState(false)}
          />
        </Modal>
        <Modal
          transparent={true}
          visible={optionsModalState}
          animationType="slide">
          <MoreBox
            options={optionsState}
            onClose={() => setOptionsModalState(false)}
            onSelect={onSelectOption}
          />
        </Modal>
        <ScrollView style={style.scrollContainer}>
          <View style={style.postNameContainer}>
            <TouchableOpacity key={post.id} onPress={navigationByCondition}>
              <View style={{flexDirection: 'row', alignItems: 'center'}}>
                <Image
                  style={style.postUserImage}
                  source={{uri: post.profileImg}}
                />
                <Text style={style.postUserName}>{post.name}</Text>
              </View>
            </TouchableOpacity>
            <View style={style.dotsDropdownContainer}>
              <View>
                <TouchableOpacity onPress={() => setOptionsModalState(true)}>
                  <Text style={style.dotsButton}>...</Text>
                </TouchableOpacity>
              </View>
            </View>
          </View>
          <Image style={[style.image]} source={{uri: post.img}} />
          <View style={style.detailsContainer}>
            <Text style={style.timestamp}>{post.timestamp}</Text>

            <View style={style.likesContainer}>
              <View style={style.iconContainer}>
                <LikeInactiveIcon />
                <Text style={style.likes}>{post.likes}</Text>
              </View>
              <View style={style.iconContainer}>
                <CommentIcon />
                <Text style={style.comments}>{post.comments}</Text>
              </View>
            </View>
          </View>
          <Text style={style.body}>{post.body}</Text>
          <UserCommentList
            postId={post.id}
            navigation={navigation}
            userAccount={userAccount}
          />
        </ScrollView>
        <View style={style.searchInputContainer}>
          <TextInput
            style={style.searchInput}
            placeholder="Enter your comment"
            placeholderTextColor="lightgrey"
          />
        </View>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
};

export default PostScreen;

const style = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    paddingVertical: moderateScale(21),
    backgroundColor: '#2a334a',
  },

  reportModalContainer: {
    flex: 1,
    marginTop: moderateScale(310),
    backgroundColor: '#3e4d6c',
    borderRadius: moderateScale(20),
    padding: moderateScale(24),
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  innerReportContainer: {
    marginTop: moderateScale(10),
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '50%',
  },
  optionModalText: {
    color: '#B8A0FF',
    marginLeft: moderateScale(16),
    fontFamily: 'Montserrat-SemiBold',
    fontSize: moderateScale(13),
    marginBottom: moderateScale(10),
  },

  scrollContainer: {
    paddingHorizontal: moderateScale(10),
  },
  image: {
    height: moderateScale(184),
    width: '100%',
    borderRadius: moderateScale(10),
  },
  postNameContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: moderateScale(9),
    justifyContent: 'space-between',
  },
  postUserImage: {
    height: moderateScale(38),
    width: moderateScale(38),
    borderRadius: moderateScale(50),
  },
  postUserName: {
    color: '#FFFFFF',
    fontSize: moderateScale(16),
    marginLeft: moderateScale(8),
    fontFamily: 'Montserrat-Bold',
  },
  detailsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: moderateScale(8),
  },
  likesContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  timestamp: {
    fontSize: moderateScale(12.5),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Italic',
  },
  iconContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  likes: {
    fontSize: moderateScale(16),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Medium',
    marginLeft: 3,
    marginRight: 14,
  },
  comments: {
    fontSize: moderateScale(16),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Medium',
    marginRight: moderateScale(1),
    marginLeft: moderateScale(3),
  },
  body: {
    fontSize: moderateScale(15),
    color: '#FFFFFF',
    marginTop: moderateScale(10),
    marginBottom: moderateScale(2),
    borderBottomWidth: moderateScale(0.7),
    borderBottomColor: 'rgba(158, 150, 150, .4)',
    paddingBottom: moderateScale(18),
    fontFamily: 'Montserrat-Medium',
  },

  dotsDropdownContainer: {
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignContent: 'center',
  },
  dotsButton: {
    alignSelf: 'flex-end',
    color: 'white',
    fontWeight: 'bold',
    fontSize: moderateScale(20),
    marginBottom: moderateScale(9),
  },
  dropdownEdit: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(1),
    marginBottom: moderateScale(-77),
    backgroundColor: '#2C3957',
    zIndex: 1,
    paddingVertical: moderateScale(6),
  },
  dropdown: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(1),
    marginBottom: moderateScale(-153),
    backgroundColor: '#2C3957',
    zIndex: 1,
    paddingVertical: moderateScale(6),
  },
  dropDownText: {
    color: 'white',
    fontSize: moderateScale(16),
    marginHorizontal: moderateScale(12),
    paddingVertical: moderateScale(2),
    fontFamily: 'Montserrat-Medium',
  },
  dropDownTextReportContainer: {
    borderTopWidth: moderateScale(1),
    borderTopColor: 'lightgrey',
    paddingTop: moderateScale(6),
    backgroundColor: '#2C3957',
  },
  searchInputContainer: {
    backgroundColor: '#2C3957',
    marginBottom: moderateScale(-10),
    paddingHorizontal: moderateScale(10),
    paddingBottom: 5,
  },
  searchInput: {
    marginTop: moderateScale(10),
    paddingLeft: moderateScale(20),
    alignContent: 'center',
    backgroundColor: '#3e4d6c',
    color: 'lightgrey',
    fontSize: moderateScale(15),
    height: moderateScale(36),
    fontFamily: 'Montserrat-Italic',
    paddingVertical: moderateScale(10),
    borderRadius: moderateScale(6),
  },
});
