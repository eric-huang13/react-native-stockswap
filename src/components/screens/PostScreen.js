import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  Image,
  SafeAreaView,
  TouchableOpacity,
  ScrollView,
  TextInput,
  Modal,
} from 'react-native';
import UserCommentList from './UserCommentList';
import ReportModal from './ReportModal';
import ShareToModal from './ShareToModal';
import LikeInactiveIcon from '../../icons/LikeInactiveIcon';
import CommentIcon from '../../icons/CommentIcon';
import { moderateScale } from '../../util/responsiveFont';

export default class PostScreen extends Component {
  constructor(props) {
    super(props);
    this.state = {
      shouldShow: false,
      reportModalState: false,
      shareModalState: false,
    };
  }
  accountId = this.props.route.params.userAccount.id;

  navigationByCondition = (post) => {
    const {navigation} = this.props;
    if (post.userId === this.accountId) {
      navigation.navigate({
        name: 'MyProfile',
        params: {id: post.id},
      });
    } else {
      navigation.navigate({
        name: 'Profile',
        params: {id: post.userId},
      });
    }
  };
  dropDownSelect(post, userAccount) {
    this.setState({shouldShow: false});
    this.props.navigation.navigate({
      name: 'EditPost',
      params: {post, userAccount},
    });
  }
  reportModal(item) {
    this.setState({reportModalState: item, shouldShow: false});
  }
  shareModal(item) {
    this.setState({shareModalState: item, shouldShow: false});
  }

  render() {
    const {shouldShow, reportModalState, shareModalState} = this.state;

    const {post, userAccount} = this.props.route.params;
    return (
      <SafeAreaView style={style.container}>
        <Modal
          transparent={true}
          visible={reportModalState}
          animationType="slide">
          <ReportModal reportModal={this.reportModal.bind(this)} />
        </Modal>
        <Modal
          transparent={true}
          visible={shareModalState}
          animationType="slide">
          <ShareToModal shareModal={this.shareModal.bind(this)} />
        </Modal>
        <ScrollView style={style.scrollContainer}>
          <View style={style.postNameContainer}>
            <TouchableOpacity
              key={post.id}
              onPress={() => this.navigationByCondition(post)}>
              <View style={{flexDirection: 'row', alignItems: 'center'}}>
                <Image
                  style={style.postUserImage}
                  source={{uri: post.profileImg}}
                />
                <Text style={style.postUserName}>{post.name}</Text>
              </View>
            </TouchableOpacity>
            {userAccount.id === post.userId ? (
              <View style={style.dotsDropdownContainer}>
                <View>
                  <TouchableOpacity
                    onPress={() =>
                      this.setState({
                        shouldShow: !shouldShow,
                      })
                    }>
                    <Text style={style.dotsButton}>...</Text>
                  </TouchableOpacity>
                </View>
                {this.state.shouldShow ? (
                  <View style={style.dropdownEdit}>
                    <TouchableOpacity
                      onPress={() => this.dropDownSelect(post, userAccount)}>
                      <Text style={style.dropDownText}>Edit post</Text>
                    </TouchableOpacity>
                    <View style={style.dropDownTextReportContainer}>
                      <Text style={style.dropDownText}>Remove post</Text>
                    </View>
                  </View>
                ) : null}
              </View>
            ) : (
              <View style={style.dotsDropdownContainer}>
                <TouchableOpacity
                  onPress={() =>
                    this.setState({
                      shouldShow: !shouldShow,
                    })
                  }>
                  <Text style={style.dotsButton}>...</Text>
                </TouchableOpacity>
                {this.state.shouldShow ? (
                  <View style={style.dropdown}>
                    <Text style={style.dropDownText}>Repost</Text>
                    <Text style={style.dropDownText}>Copy link</Text>
                    <TouchableOpacity onPress={() => this.shareModal(true)}>
                      <Text style={style.dropDownText}>Share to...</Text>
                    </TouchableOpacity>
                    <Text style={style.dropDownText}>
                      Turn on notifications
                    </Text>
                    <TouchableOpacity
                      key={post.id}
                      onPress={() => this.reportModal(true)}>
                      <View style={style.dropDownTextReportContainer}>
                        <Text style={style.dropDownText}>Report</Text>
                      </View>
                    </TouchableOpacity>
                  </View>
                ) : null}
              </View>
            )}
          </View>
          <Image style={style.image} source={{uri: post.img}} />
          <View style={style.detailsContainer}>
            <Text style={style.timestamp}>{post.timestamp}</Text>

            <View style={style.likesContainer}>
              <View style={style.iconContainer}>
                <LikeInactiveIcon />
                <Text style={style.likes}>{post.likes}</Text>
              </View>
              <View style={style.iconContainer}>
                <CommentIcon />
                <Text style={style.comments}>{post.comments}</Text>
              </View>
            </View>
          </View>
          <Text style={style.body}>{post.body}</Text>
          <UserCommentList
            postId={post.id}
            navigation={this.props.navigation}
            userAccount={userAccount}
          />
        </ScrollView>
        <View style={style.searchInputContainer}>
          <TextInput
            style={style.searchInput}
            placeholder="Enter your comment"
            placeholderTextColor="lightgrey"
          />
        </View>
      </SafeAreaView>
    );
  }
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    paddingVertical: moderateScale(21),
    backgroundColor: '#2a334a',
  },

  reportModalContainer: {
    flex: 1,
    marginTop: moderateScale(310),
    backgroundColor: '#3e4d6c',
    borderRadius: moderateScale(20),
    padding: moderateScale(24),
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  innerReportContainer: {
    marginTop: moderateScale(10),
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '50%',
  },
  optionModalText: {
    color: '#B8A0FF',
    marginLeft: moderateScale(16),
    fontFamily: 'Montserrat-SemiBold',
    fontSize: moderateScale(13),
    marginBottom: moderateScale(10),
  },

  scrollContainer: {
    paddingHorizontal: moderateScale(10),
  },
  image: {
    height: moderateScale(184),
    width: '100%',
    borderRadius: moderateScale(10),
  },
  postNameContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: moderateScale(9),
    justifyContent: 'space-between',
  },
  postUserImage: {
    height: moderateScale(38),
    width: moderateScale(38),
    borderRadius: moderateScale(50),
  },
  postUserName: {
    color: '#FFFFFF',
    fontSize: moderateScale(16),
    marginLeft: moderateScale(8),
    fontFamily: 'Montserrat-Bold',
  },
  detailsContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginTop: moderateScale(8),
  },
  likesContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
  timestamp: {
    fontSize: moderateScale(12.5),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Italic',
  },
  iconContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  likes: {
    fontSize: moderateScale(16),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Medium',
    marginLeft: 3,
    marginRight: 14,
  },
  comments: {
    fontSize: moderateScale(16),
    color: 'lightgrey',
    fontFamily: 'Montserrat-Medium',
    marginRight: moderateScale(1),
    marginLeft: moderateScale(3),
  },
  body: {
    fontSize: moderateScale(15),
    color: '#FFFFFF',
    marginTop: moderateScale(10),
    marginBottom: moderateScale(2),
    borderBottomWidth: moderateScale(0.7),
    borderBottomColor: 'rgba(158, 150, 150, .4)',
    paddingBottom: moderateScale(18),
    fontFamily: 'Montserrat-Medium',
  },

  dotsDropdownContainer: {
    flexDirection: 'column',
    justifyContent: 'space-evenly',
    alignContent: 'center',
  },
  dotsButton: {
    alignSelf: 'flex-end',
    color: 'white',
    fontWeight: 'bold',
    fontSize: moderateScale(20),
    marginBottom: moderateScale(9),
  },
  dropdownEdit: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(1),
    marginBottom: moderateScale(-77),
    backgroundColor: '#2C3957',
    zIndex: 1,
    paddingVertical: moderateScale(6),
  },
  dropdown: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'space-between',
    width: '100%',
    marginTop: moderateScale(1),
    marginBottom: moderateScale(-153),
    backgroundColor: '#2C3957',
    zIndex: 1,
    paddingVertical: moderateScale(6),
  },
  dropDownText: {
    color: 'white',
    fontSize: moderateScale(16),
    marginHorizontal: moderateScale(12),
    paddingVertical: moderateScale(2),
    fontFamily: 'Montserrat-Medium',
  },
  dropDownTextReportContainer: {
    borderTopWidth: moderateScale(1),
    borderTopColor: 'lightgrey',
    paddingTop: moderateScale(6),
    backgroundColor: '#2C3957',
  },
  searchInputContainer: {
    backgroundColor: '#2C3957',
    marginBottom: moderateScale(-10),
    paddingHorizontal: moderateScale(10),
  },
  searchInput: {
    marginTop: moderateScale(10),
    paddingLeft: moderateScale(20),
    alignContent: 'center',
    backgroundColor: '#3e4d6c',
    color: 'lightgrey',
    fontSize: moderateScale(15),
    height: moderateScale(36),
    fontFamily: 'Montserrat-Italic',
    paddingVertical: moderateScale(10),
    borderRadius: moderateScale(6),
  },
});
