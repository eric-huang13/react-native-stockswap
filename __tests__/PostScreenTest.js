import React from 'react';
import PostScreen from '../src/components/HomeTabScreens/PostScreen';
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));
  

test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render   (<PostScreen route={{params: {userAccount:{id:'test',}, post:{id:'test'}}}}/>);
    const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});