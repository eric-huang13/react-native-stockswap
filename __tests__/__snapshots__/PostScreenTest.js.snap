// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#2a334a",
      "flex": 1,
      "flexDirection": "column",
      "paddingVertical": 29.66058394160584,
    }
  }
>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <ReportModal
      reportModal={[Function]}
    />
  </Component>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  >
    <ShareToModal
      shareModal={[Function]}
    />
  </Component>
  <ScrollView
    style={
      Object {
        "paddingHorizontal": 14.124087591240876,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginBottom": 12.711678832116789,
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "borderRadius": 70.62043795620437,
                "height": 53.67153284671533,
                "width": 53.67153284671533,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 22.598540145985403,
                "marginLeft": 11.299270072992702,
              }
            }
          />
        </View>
      </ForwardRef>
      <View
        style={
          Object {
            "alignContent": "center",
            "flexDirection": "column",
            "justifyContent": "space-evenly",
          }
        }
      >
        <ForwardRef
          onPress={[Function]}
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "white",
                "fontSize": 28.248175182481752,
                "fontWeight": "bold",
                "marginBottom": 12.711678832116789,
              }
            }
          >
            ...
          </Text>
        </ForwardRef>
      </View>
    </View>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 14.124087591240876,
          "height": 259.88321167883214,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 11.299270072992702,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "lightgrey",
            "fontFamily": "Montserrat-Italic",
            "fontSize": 17.655109489051092,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <LikeInactiveIcon />
          <Text
            style={
              Object {
                "color": "lightgrey",
                "fontFamily": "Montserrat-Medium",
                "fontSize": 22.598540145985403,
                "marginLeft": 3,
                "marginRight": 14,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <CommentIcon />
          <Text
            style={
              Object {
                "color": "lightgrey",
                "fontFamily": "Montserrat-Medium",
                "fontSize": 22.598540145985403,
                "marginLeft": 4.237226277372263,
                "marginRight": 1.4124087591240877,
              }
            }
          />
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "borderBottomColor": "rgba(158, 150, 150, .4)",
          "borderBottomWidth": 0.9886861313868612,
          "color": "#FFFFFF",
          "fontFamily": "Montserrat-Medium",
          "fontSize": 21.186131386861312,
          "marginBottom": 2.8248175182481754,
          "marginTop": 14.124087591240876,
          "paddingBottom": 25.423357664233578,
        }
      }
    />
    <UserCommentList
      postId="test"
      userAccount={
        Object {
          "id": "test",
        }
      }
    />
  </ScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#2C3957",
        "marginBottom": -14.124087591240876,
        "paddingHorizontal": 14.124087591240876,
      }
    }
  >
    <Component
      allowFontScaling={true}
      placeholder="Enter your comment"
      placeholderTextColor="lightgrey"
      rejectResponderTermination={true}
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#3e4d6c",
          "borderRadius": 8.474452554744525,
          "color": "lightgrey",
          "fontFamily": "Montserrat-Italic",
          "fontSize": 21.186131386861312,
          "height": 50.846715328467155,
          "marginTop": 14.124087591240876,
          "paddingLeft": 28.248175182481752,
          "paddingVertical": 14.124087591240876,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
</ForwardRef(SafeAreaView)>
`;
